### PR TITLE
Updates hardware readmes

### PR DESCRIPTION
**Pull request by KipHamiltons**
These hardware readmes are out of date. This PR updates them by changing
- namespaces on the messages
- the robot's name (Darwin -> NUgus)
- removing references to emitting vectors of messages, which we don't do any more

### CODE REVIEW

**Review by TrentHouliston — CHANGES_REQUESTED** [review]
The namespace is still darwin.

IMO this isn't a "NUgus" or "Darwin" platform
Instead I would rename it to CM740, since that's actually the platform that this is connecting to

File: module/platform/darwin/HardwareSimulator/README.md [hunk 1]
```diff
@@ -1,40 +1,36 @@
-Darwin Hardware I/O
+NUgus Hardware I/O
 ===================
 
 ## Description
 
-This module is responsible for communicating with the Darwin robot's CM740
-controller.
+This module is responsible for communicating with the NUgus' CM740 controller.
```
Comment (ysims) [inline module/platform/darwin/HardwareSimulator/README.md:6]: If we're updating this might as well fix it so it's not a copy of the HardwareIO readme. This module doesn't connect to the CM740 (or any device), instead, it simulates all the sensor info we want.

File: module/platform/darwin/HardwareIO/README.md [hunk 2]
```diff
@@ -1,40 +1,36 @@
...
 
-This module is responsible for communicating with the Darwin robot's CM740
-controller.
+This module is responsible for communicating with the NUgus' CM740 controller.
 
 ## Usage
 
-Darwin Hardware I/O connects at startup to the Darwin controller located on
+NUgus Hardware I/O connects at startup to the NUgus controller located on
 `/dev/usbTTY0`. If this does not succeed an exception is thrown and startup is
```
Comment (Bidski) [inline module/platform/darwin/HardwareIO/README.md:11]: [udev rules](https://github.com/NUbots/NUbots/blob/b411e14fa76bf36d742531cb83d42bb7a42490d3/doc/ArchInstall/arch-chroot_install.sh#L205-L216) are set up on the robot to automagically map the CM740 device to `/dev/CM740`
```suggestion
`/dev/CM740`. If this does not succeed an exception is thrown and startup is
```

File: module/platform/darwin/HardwareIO/README.md [hunk 3]
```diff
@@ -1,40 +1,36 @@
...
 aborted.
 
-This module reads the current status of the Darwin 50 times per second and
-emits it as a `message::DarwinSensors` object. This includes the CM740 error
+This module reads the current status of the NUgus 50 times per second and
```
Comment (Bidski) [inline module/platform/darwin/HardwareIO/README.md:14]: See [UPDATE_FREQUENCY](https://github.com/NUbots/NUbots/blob/b411e14fa76bf36d742531cb83d42bb7a42490d3/module/platform/darwin/HardwareIO/src/HardwareIO.hpp#L41)
```suggestion
This module reads the current status of the NUgus 90 times per second and
```

File: module/platform/darwin/HardwareSimulator/README.md (hunk 1 above)
Comment (Bidski) [reply module/platform/darwin/HardwareSimulator/README.md:6]: Also, see my suggestion for the README in this PR

**Review by Bidski — CHANGES_REQUESTED** [review]

File: module/platform/darwin/HardwareSimulator/README.md (hunk 1 above)
Comment (KipHamiltons) [reply module/platform/darwin/HardwareSimulator/README.md:6]: > Also, see my suggestion for the README in this PR
was this meant to be a link?

**Comment by KipHamiltons** [comment]
> The namespace is still darwin.
> 
> IMO this isn't a "NUgus" or "Darwin" platform
> Instead I would rename it to CM740, since that's actually the platform that this is connecting to

As in, rename the namespace itself, and the title of the README? So like, the `CM740 Hardware I/O` readme?

**Comment by TrentHouliston** [comment]
Yep

File: module/platform/darwin/HardwareSimulator/README.md (hunk 1 above)
Comment (KipHamiltons) [reply module/platform/darwin/HardwareSimulator/README.md:6]: @Bidski could you please clarify what you meant

Comment (Bidski) [reply module/platform/darwin/HardwareSimulator/README.md:6]: @KipHamiltons look at the PR review comments I made the HardwareIO README.md file

Comment (KipHamiltons) [reply module/platform/darwin/HardwareSimulator/README.md:5]: Ahh I understand. thanks

**Comment by KipHamiltons** [comment]
I have been putting off updating this PR because `darwin` changes will mean some of this will have to be changed anyway. Since I haven't got around to making those changes I have made an issue: #758

**Comment by ysims** [comment]
This should be good now.

File: module/input/SensorFilter/src/SensorFilter.cpp [hunk 4]
```diff
@@ -487,8 +487,6 @@ namespace module::input {
                         sensors->led.emplace_back(0, input.led_panel.led2 ? 0xFF0000 : 0);
                         sensors->led.emplace_back(1, input.led_panel.led3 ? 0xFF0000 : 0);
                         sensors->led.emplace_back(2, input.led_panel.led4 ? 0xFF0000 : 0);
-                        sensors->led.emplace_back(3, input.head_led.RGB);  // Head
-                        sensors->led.emplace_back(4, input.eye_led.RGB);   // Eye
```
Comment (Bidski) [inline module/input/SensorFilter/src/SensorFilter.cpp:491]: This has nothing to do with the READMEs?

File: module/platform/cm740/HardwareIO/README.md [hunk 5]
```diff
@@ -13,20 +12,14 @@ aborted.
 
 This module reads the current status of the CM740 90 times per second and
 emits it as a `message::platform::RawSensors` object. This includes the CM740 error
-code, LED panel, head and eye LED colour, buttons, voltage, accelerometer,
-gyroscope, left and right force-sensing resistors and each servo.
-
-To change the colour of the NUgus' head or eye LEDs, emit a
-`message::platform::RawSensors::EyeLED` or `message::platform::RawSensors::HeadLED`
-containing the colour you wish to set them to.
```
Comment (Bidski) [inline module/platform/cm740/HardwareIO/README.md:21]: The functionality still exists (the CM740 still has the hardware connections available to connect up these LEDs), not the modules fault that we don't take advantage of it.

File: module/platform/cm740/HardwareIO/src/HardwareIO.hpp [hunk 6]
```diff
@@ -37,69 +37,77 @@ namespace module::platform::cm740 {
...
+            /// @brief Current angle position of the servo. Simulated or read.
+            float present_position = 0;
+            /// @brief Current speed of the servo. Simulated or read.
+            float present_speed = 0;
+            /// @brief Current load of the servo. Simulated or read.
+            float load = 0;
+            /// @brief Current voltage of the servo. Simulated or read.
+            float voltage = 10;
+            /// @brief Current temperature of the servo. Simulated or read.
+            float temperature = 40;
```
Comment (Bidski) [inline module/platform/cm740/HardwareIO/src/HardwareIO.hpp:106]: technically you are initialising `float`s with `int`s here .... although the scope creep in this PR is already pretty insane

File: module/platform/cm740/HardwareSimulator/src/HardwareSimulator.cpp [hunk 7]
```diff
@@ -62,12 +62,6 @@ namespace module::platform::cm740 {
         sensors.led_panel.led3 = false;
         sensors.led_panel.led4 = false;
 
-        // Head LED
-        sensors.head_led.RGB = 0;
-
-        // Head LED
-        sensors.eye_led.RGB = 0;
-
```
Comment (Bidski) [inline module/platform/cm740/HardwareSimulator/src/HardwareSimulator.cpp:70]: I think these should remain along with the LED fields in `HardwareIO`

**Review by Bidski — CHANGES_REQUESTED** [review]
The scope creep in this PR is nuts. 90% of the changes have nothing to do with updating the READMEs

**Comment by ysims** [comment]
Probably right @Bidski I just was hoping to get some cleaning up changes in too. I've moved it to a different branch and added back in the LEDs, with a mention in here about the nugus not having it but having the capability through the cm740, so people aren't confused.

File: module/platform/cm740/HardwareIO/README.md [hunk 8]
```diff
@@ -12,10 +12,15 @@ aborted.
 
 This module reads the current status of the CM740 90 times per second and
 emits it as a `message::platform::RawSensors` object. This includes the CM740 error
-code, LED panel, buttons, voltage, accelerometer, gyroscope, left and right
-force-sensing resistors and each servo.
+code, LEDs, buttons, voltage, accelerometer, gyroscope, left and right
+force-sensing resistors and each servo. Darwin robots have head and eye LEDs which are
+referred to in this module and can be read from the CM740,
+however the NUgus robot does not utilise this capability.
```
Comment (Bidski) [inline module/platform/cm740/HardwareIO/README.md:18]:
```suggestion
force-sensing resistors and each servo. CM740s have the ability to connect to head and eye LEDs and force-sensing resistors which are
referred to in this module and can be read from the CM740,
however the NUgus robot does not utilise these capabilities at present.
```

**Review by Bidski — CHANGES_REQUESTED** [review]
Seems a lot better now

**Comment by ysims** [comment]
Way too out of date now, cm740 is gone.
